src/modules/ethereum/exchangeTransactions.js

```diff
@@ -211,6 +182,111 @@ export async function matchOrdersTx(buyId, sellId) {
     return { txName, transactionHash };
 }
 
+export async function matchMultipleOrdersTx() {
+    const txName = "Match orders";
+    const userAccount = store.getState().web3Connect.userAccount;
+    const exchange = store.getState().contracts.latest.exchange.web3ContractInstance;
+    const bn_ethFiatRate = store.getState().rates.info.bn_ethFiatRate;
+    const orders = store.getState().orders.orders;
+    const lowestSellPrice = orders.sellOrders[0].price;
+    const highestBuyPrice = orders.buyOrders[0].price;
+
+    const buyOrders = orders.buyOrders
+        .filter(o => o.price >= lowestSellPrice)
+        .map(o => ({ id: o.id, price: o.price, bn_ethAmount: o.bn_ethAmount }));
+    const sellOrders = orders.sellOrders
+        .filter(o => o.price <= highestBuyPrice)
+        .map(o => ({ id: o.id, price: o.price, bn_tokenAmount: new BigNumber(o.amount) }));
+
+    const sellIds = [];
+    const buyIds = [];
+    let buyIdx = 0;
+    let sellIdx = 0;
+    let matchCount = 0;
+    while (
+        buyIdx < buyOrders.length &&
+        sellIdx < sellOrders.length &&
+        buyOrders[buyIdx].price >= sellOrders[sellIdx].price && // we already filtered but just in case...
+        matchCount < 100 // to make sure gas cost won't be over block gas limit even in edge scenarios
```

Comment:
Calculate the gas estimate on the fly, with an explicit "gas limit" parameter (that could be set by the caller).

(if you need the number of matches later, just use `sellIds.length`)

Reply:
using `sellIds.length` is a good call.
i don't understand rest about gas limit, what do you mean?

Reply:
I mean that there should be a `gasLimit` parameter, and as you iterate you should tally a running total of the estimated gas costs and bail from the loop once you have reached it (rather than computing the estimate at the end)

Reply:
It might be an overkill for an edge case (and also I couldn't accurately estimate the gas cost for each additional match).  
i think it's much simpler that we just match max 100 orders, if there is more than click again... what's the chance to have more than 100? Especially if we will have the new auto matcher immediately executing matches.

Reply:
my point is: I would make the `limit` a parameter to the method (currently hardcoded as 100) and specify it in gas rather than number of matches.

Reply:
Ok, I got ya. I suggest to keep match limit b/c that's much understandable than gas limit. Gas limit you can always adjust in Metamask having the same effect. 
Also gas cost is only an estimate, i.e. we estimate in FE that 10 matches will fit to 500k gas but it can easily end up less and additional matches would have fit in.
Again, it is an edge case I can't see any problem leaving as it is

Reply:
Specifying the gas limit in Metamask will not help, since the tx is already built by then (without consideration for this limit)

Reply:
What is the scenario / use case / issue you want to tackle with the suggested change?

Reply:
Improved code quality. The following would clearly express the intent why the exit condition and the gas estimation code is there in the first place, instead of a magic constant and a comment:

```suggestion
        estimatedGas < gasLimit
```